Tests/DataPipeline/Support/TestUtilities.swift

```diff
@@ -76,12 +77,24 @@ extension RawEvent {
 
 // MARK: - Helper Methods
 
-func waitUntilStarted(analytics: Analytics?) {
+func waitUntilStarted(analytics: Analytics? = DataPipeline.shared.analytics) {
     guard let analytics = analytics else { return }
+
     // wait until the startup queue has emptied it's events.
     if let startupQueue = analytics.find(pluginType: StartupQueue.self) {
         while startupQueue.running != true {
             RunLoop.main.run(until: Date.distantPast)
         }
     }
 }
+
+/// Attaches and returns plugin only if it wasn't attached previously
+func attachPlugin<P: Plugin>(analytics: Analytics? = DataPipeline.shared.analytics, plugin: P) -> P {
+    guard let analytics = analytics else { return plugin }
```

Comment:
```suggestion
    guard let analytics = analytics else { fatalError() }
```

I was not expecting to see this line of code exit early and ignore the request. Is there a reason to do this instead of fail?  

If a developer calls this function to attach a plugin, the function runs without throwing an error, they would expect the plugin got attached but maybe it did not. I would expect that by ignoring the request, this could make writing and debugging tests more difficult.

Reply:
Updated 👍🏻